tools/wasi/wasi.cpp

```diff
@@ -152,32 +150,25 @@ bool run(bytes_view wasm_binary, int argc, const char* argv[], std::ostream& err
         {ns, "environ_get", {ValType::i32, ValType::i32}, ValType::i32, return_enosys},
     };
 
-    // Initialisation settings.
-    // TODO: Make const after https://github.com/nodejs/uvwasi/pull/155 is merged.
-    uvwasi_options_t options = {
-        3,           // sizeof fd_table
-        0, nullptr,  // NOTE: no remappings
-        static_cast<uvwasi_size_t>(argc), argv,
-        nullptr,  // TODO: support env
-        0, 1, 2,
-        nullptr,  // NOTE: no special allocator
-    };
+    auto module = parse(wasm_binary);
+    auto imports = resolve_imported_functions(*module, wasi_functions);
+    return instantiate(std::move(module), std::move(imports), {}, {}, {}, MaxMemoryPagesLimit);
+}
 
-    const uvwasi_errno_t uvwasi_err = uvwasi_init(&state, &options);
+bool run(Instance& instance, int argc, const char* argv[], std::ostream& err)
+{
+    if (!uvwasi)
```

Comment:
I don't think this is good idea, for many reasons (e.g. not thread-safe, hidden dependencies). Why just not add `UVWASI&` param?

Reply:
This is a good idea.

Do you want also to pass it into WASI functions via `host_context` to make it thread-safe?

Reply:
I don't see why this would be needed. This object will be bound to the imported functions, right?
By "not-thread-safe" I mean the initialization. I think this should be propagated up to the `main()` function which should invoke `create_uvwasi()`. And `fizzy-unittest-wasi` will execute `create_mocked_uvwasi()` if needed.

Reply:
> I don't see why this would be needed. This object will be bound to the imported functions, right?

What do you mean by bound? The way to bind it in current API is to pass it via `host_context`.

See the latest version.

Reply:
It looks good.